Multiplatform/Shared/Sidebar/RefreshProgressModel.swift

```diff
@@ -0,0 +1,101 @@
+//
+//  RefreshProgressModel.swift
+//  NetNewsWire
+//
+//  Created by Phil Viso on 7/2/20.
+//  Copyright © 2020 Ranchero Software. All rights reserved.
+//
+
+import Account
+import Combine
+import Foundation
+import SwiftUI
+
+class RefreshProgressModel: ObservableObject {
+	
+	enum State {
+		case refreshProgress(Float)
+		case lastRefreshDateText(String)
+		case none
```

Comment:
I specifically added a `none` case here instead of having the `state` variable be optional, because I didn't want the view to need to interpret what a nil state meant. I'm not sure it's even technically possible. When the app first launches a feed refresh is triggered, and whether it succeeds or fails we always end up we a last refresh date right after that. We should always be in a state of having `refreshProgress` or `lastRefreshDateText`.